infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/events/BrokerStatusListener.java

```diff
@@ -60,20 +64,25 @@ public void onEvent(BrokerEvent event) {
         if (tooling != null) {
           try {
             pluginsValidator.validatePluginNames(tooling);
-            finishFuture.complete(tooling);
           } catch (ValidationException e) {
-            finishFuture.completeExceptionally(e);
+            brokersResult.error(e);
+            return;
+          }
+          try {
+            brokersResult.brokerResult(tooling);
```

Comment:
`brokersResult.brokerResult` looks a bit unclear. Please consider renaming to something like `addResult`, `registerResult`

Reply:
ok